File: grupos/gruposEspeciais.js
const texto = 'João é calmo, mas no transito fica nervoso.'

console.log(texto.match(/[\wÁ-ú]+(?=,)/gi))//lookahead positive

//negative lookahead

console.log(texto.match(/[\wÀ-ú]+\b(?!,)/gi))
console.log(texto.match(/[\wÀ-ú]+[\s|\.](?!,)/gi))

File: grupos/gruposEspeciais.rb
texto = 'supermercado superação hiperMERCADO'

puts texto.scan(/(?:super)[\wÀ-ú]+/i).join(' ')

# Positive Lookbehind
puts texto.scan(/(?<=super)[\wÀ-ú]+/i).join(' ')

# Negative lookbehind

puts texto.scan(/(?<!super)mercado/i)